record/record.go

```diff
@@ -158,9 +160,17 @@ func (r *Reader) nextChunk(wantFirst bool) error {
 			r.i = r.j + headerSize
```

Comment:
Do we also need to recover with the `return errors.New("leveldb/record: invalid chunk")` a few lines above?